src/common/context/forms/index.tsx

```diff
@@ -64,25 +65,34 @@ export const FormsContextProvider: FunctionComponent = ({ children }) => {
     setActiveFormIndex(newIndex);
   };
 
-  const newPopulatedForm = (templateIndex: number, data: Array<FormEntry>): void => {
-    const newFormTemplate = config?.forms[templateIndex];
-    const newFormName = newFormTemplate?.name ?? "Document";
-    const formEntries: FormEntry[] = [];
-    const extension = config?.forms[templateIndex]?.extension ?? "tt";
-    for (let index = 0; index < data.length; index++) {
-      formEntries.push({
-        templateIndex,
-        data: {
-          formData: data[index],
-          schema: newFormTemplate?.schema,
-        },
-        fileName: data[index].fileName ?? `${newFormName}-${forms.length + 1 + index}`,
-        ownership: data[index].ownership ?? { beneficiaryAddress: "", holderAddress: "" },
-        extension: extension,
-      });
+  const newPopulatedForm = (templateIndex: number, data: Array<FormEntry>, fileName?: string): void => {
+    try {
+      const newFormTemplate = config?.forms[templateIndex];
+      const newFormName = newFormTemplate?.name ?? "Document";
+      const formEntries: FormEntry[] = [];
+      const extension = config?.forms[templateIndex]?.extension ?? "tt";
+      const filenameTemplate = fileName ? _.template(fileName) : undefined;
+
+      for (let index = 0; index < data.length; index++) {
+        formEntries.push({
+          templateIndex,
+          data: {
+            formData: data[index],
+            schema: newFormTemplate?.schema,
+          },
+          fileName: filenameTemplate ? filenameTemplate(data[index]) : `${newFormName}-${forms.length + 1 + index}`,
```

Comment:
currently the code looks to be generating 2 fileName,
1) is the newFormName
2) is the filenameTemplate that u created

Able to refactor them into a function that will generate 1 fileName only?